work/v1/types.go

```diff
@@ -66,6 +70,65 @@ type DeleteOption struct {
 	SelectivelyOrphan *SelectivelyOrphan `json:"selectivelyOrphans,omitempty"`
 }
 
+// ManifestConfigOption represents the configurations of a manifest defined in workload field.
+type ManifestConfigOption struct {
+	// ResourceIdentifier represents the group, resource, name and namespace of a resoure.
+	// +kubebuilder:validation:Required
+	// +required
+	ResourceIdentifier ResourceIdentifier `json:"resourceIdentifier"`
+
+	// FeedbackRules defines what resource status field should be returned.
+	// +kubebuilder:validation:Required
+	// +required
+	FeedbackRules []FeedbackRule `json:"feedbackRules"`
+}
+
+type FeedbackRule struct {
+	// Type defines the option of how status can be returned.
+	// It can be jsonPaths only or wellKnownStatus.
+	// If the type is JsonPaths, user should specify the jsonPaths field
+	// If the type is WellKnownStatus, certain common fields of status will be reported,
+	// including replicas, readyReplicas, and availableReplicas. And if these status fields
+	// do not exist, no values will be reported.
+	// +kubebuilder:validation:Required
+	// +required
+	Type FeedBackType `json:"type"`
+
+	// JsonPaths defines the json path under status field to be synced.
+	// +optional
+	JsonPaths []JsonPath `json:"jsonPaths,omitempty"`
+}
+
+// +kubebuilder:validation:Enum=WellKnownStatus;JsonPaths
+type FeedBackType string
```

Comment:
we can have another feedback type like `StatusCondition`, which will be working like:

```
manifestConfigs: 
 - resourceIdentifier: 
     resource: Foo 
     version: v1 
     group: FooGroup 
     name: foo 
   feedbackRules: 
   - type: StatusCondition
     statusCondition:
       conditionType: MyConditionType    
...
resourceStatus:
  manifests:
  - resourceMeta:  ...
    statusFeedback:
      values:
      - name: foo
         fieldValue: 
           type: Condition
           condition: ...    
```

Reply:
with json path, you can have

```yaml
- type: StatusJsonPaths
      jsonPaths:
      - name: availableCondition
        path: .conditions[?(@.type=="Available")].status 
```

will it be good enough?

Reply:
IIRC we dont need a new feedback type `StatusJsonPaths` if it's describe'ble via a json path, is it?

cc @zychina for evaluating the usability.